### ISSUE TITLE
Global focus indicator styles. Final UI design to integrate to Style Cards Manager UI (SCM).

### ISSUE THREAD

**Issue by ckp267** [maxi-blocks/maxi-blocks#2726]
### Context
Basic assessability requires a visual **focus indicator** for visitors who use keyboards to navigate via tab/enter. Also via voice. (https://www.w3.org/TR/WCAG22/) 

### What is a focus indicator?
A focus indicator is a visual indicator that “highlights” the currently focused element. This visual indicator is commonly presented as an outline around the element. An outline takes the shape of its element, and since every element in CSS is a rectangle, an outline is, therefore, usually a rectangle drawn around an element.

A focus indicator can also take other forms, but outlines are very common for several reasons. 

- Outlines have an advantage over other visual indicators (such as borders or background colours, for example) in that they can be applied to the element without causing any significant changes to that element. 
- And since an outline is not part of an element’s box model, it does not affect the layout of that element, and will therefore not cause any layout shifts when it is applied. (That’s also why outlines are preferred over borders for visualizing and debugging layouts 💡) 
- In addition to that, outlines are preserved and used in Forced Colour Modes such as Windows High Contrast Mode, where background colours, border colours, and box shadows are usually overridden by user and system styles.

So a focus indicator allows a keyboard user to see exactly where they are at any given moment. Without it, they wouldn’t know where they are on a page and they wouldn’t be able to navigate the page and operate its controls.

**The focus indicator is to keyboard users what the mouse cursor is to mouse users.** And just like you would never want to hide the mouse cursor, you never want to hide the focus indicator.

### Why is it relevant for Maxi Blocks?

1. To make all Maxi Blocks built websites accessible. Good for visitors, good for builders, good for us. More inclusive - everyone has a legal right to be able to use a website. 
2. To reduce Maxi Blocks support load. We need a simple, adjustable solution for the focus indicator that works on all blocks.
3. It's the right thing to do for the long term growth of Maxi Blocks. We will get loyalty from people who use our product. 

NOTE: Previously we introduced accessibility focus indicators to the Maxi Blocks Editor UI. It is a plugin setting. This is not what we are discussing here. This is a separate issue. 

**Future additions to consider for V2 of Maxi Blocks**

- Pressed indicator for touch screens. Also called the ripple effect. Simulates a mouse click to communicate an accurate tap on a touch screen. https://uxmovement.com/mobile/the-hover-effect-for-mobile-buttons/ 
https://codepen.io/finnhvman/post/pure-css-ripple-with-minimal-effort
- Applies only to the Pressed state. A pressed state communicates a user tap.

### Proposed solution

FYI: Material design UI focus indicators fail the contrast test (https://www.w3.org/TR/WCAG22/). Bootstrap 4 buttons also fail WCAG requirements because of low change in contrast between the focused and unfocused states.

Requirements: (this has already been discussed and approved with @elzadj )

1. Show the focus indicators only for keyboard users on tab. (not when you click with a mouse pointer)
You can do that using the `:focus-visible` pseudo-class. :focus-visible does exactly the same thing `:focus `does, except that it only applies the focus indicator styles to an element when that element receives focus via keyboard. More info here: https://www.sarasoueidan.com/blog/focus-indicators/ 
2. Make focus indicator style an outline with an `offset -6` so that it sits inside of the element. Gutenberg does it like this. We will keep ours similar and familiar. 
3. Allow user control of line type, thickness, outline offset and line colour. Line colour will have 2 variants specifically for Light tone and Dark tone blocks. Will provide the necessary contrast.
4. Accessibility focus indicator will be a integrated to Style Card Manager. It requires components for
- line type,
- thickness, 
- colour,
- and outline offset 
- light tone and dark tone variant colour options (2 colour choices total)
6. The chosen styles will be implemented via Style Card variable and applied automatically to blocks via Light tone and Dark tone selectors.
